### PR TITLE
[Mobile] Budget menu modals

### PR DESCRIPTION
**Pull request by joel-jeremy**
- Removed Add group button and replaced with a + sign on page header
- Added a menu on the left side of the page header for more budget menus
  - (Will add budget actions e.g. goal template menus next)

### CODE REVIEW

File: upcoming-release-notes/2491.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+category: Enhancements
+authors: [joel-jeremy]
+---
+
+Mobile budget page menu modals.
```
Comment (MatissJanis) [inline upcoming-release-notes/2491.md:6]: **🥜 nitpick:** ‏a more detailed description would be better

File: packages/desktop-client/src/components/modals/CategoryGroupMenuModal.tsx [hunk 2]
```diff
@@ -91,7 +91,9 @@ export function CategoryGroupMenuModal({
 
   return (
     <Modal
-      title={group.name}
+      title={
+        <ModalTitle isEditable title={group.name} onTitleUpdate={onRename} />
+      }
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/modals/CategoryGroupMenuModal.tsx:96]: Could we remove left-padding from the mobile modal title?

Screenshot 2024-04-06 at 19 51 10

**Review by MatissJanis — DISMISSED** [review]
Overall LGTM

File: packages/desktop-client/src/components/modals/CategoryGroupMenuModal.tsx (hunk 2 above)
Comment (joel-jeremy) [reply packages/desktop-client/src/components/modals/CategoryGroupMenuModal.tsx:96]: The modal title is centered. That left margin is reserved for any additional menus e.g. category modal's hide and delete menus.

**Review by MatissJanis — DISMISSED** [review]

**Review by shall0pass — APPROVED** [review]

**Pull request merged**